src/meson.build

```diff
@@ -113,27 +116,45 @@ if libexif_enabled
   nemo_deps += libexif
 endif
 
+
+nemo_link_args = []
+if use_jemalloc
+  nemo_link_args += ['-Wl,--no-as-needed', '-ljemalloc', '-Wl,--as-needed']
+endif
 nemo = executable('nemo',
   nemoCommon_sources + nemoWindow_sources,
   include_directories: [ rootInclude ],
   c_args: nemo_definitions,
   dependencies: nemo_deps,
+  link_args: nemo_link_args,
   install: true
 )
 
+
+nemoDesktop_link_args = []
+if use_jemalloc
+  nemoDesktop_link_args += ['-Wl,--no-as-needed', '-ljemalloc', '-Wl,--as-needed']
+endif
 nemoDesktop = executable('nemo-desktop',
   nemoCommon_sources + nemoDesktop_sources,
   include_directories: [ rootInclude],
   c_args: nemo_definitions,
   dependencies: nemo_deps,
+  link_args: nemoDesktop_link_args,
   install: true
 )
 
+
+nemo_autorun_software_link_args = []
+if use_jemalloc
+  nemo_autorun_software_link_args += ['-Wl,--no-as-needed', '-ljemalloc', '-Wl,--as-needed']
+endif
 nemo_autorun_software = executable('nemo-autorun-software',
   [ 'nemo-autorun-software.c' ],
   include_directories: [ rootInclude, ],
   c_args: nemo_definitions,
   dependencies: nemo_deps,
+  link_args: nemo_autorun_software_link_args,
   install: true
 )
```

Comment:
I don't think this needs to use jemalloc, it's just a Gtk dialog for when media gets inserted.